apps/api/v1/pages/api/connected-calendars/_get.ts

```diff
@@ -0,0 +1,71 @@
+import type { NextApiRequest } from "next";
+
+import type { UserWithCalendars } from "@calcom/lib/getConnectedDestinationCalendars";
+import { getConnectedDestinationCalendars } from "@calcom/lib/getConnectedDestinationCalendars";
+import { HttpError } from "@calcom/lib/http-error";
+import { defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+import { extractUserIdsFromQuery } from "~/lib/utils/extractUserIdsFromQuery";
+
+/**
+ * @swagger
+ * /connected-calendars:
+ *   get:
+ *     parameters:
+ *       - in: query
+ *         name: apiKey
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: Your API key
+ *       - in: query
+ *         name: userId
+ *         required: false
+ *         schema:
+ *           type: number
+ *         description: Admins can fetch connected calendars for other user e.g. &userId=1 or multiple users e.g. &userId=1&userId=2
+ *     summary: Fetch connected calendars
+ *     tags:
+ *      - connected-calendars
+ *     responses:
+ *       200:
+ *         description: OK
+ *       401:
+ *        description: Authorization information is missing or invalid.
+ *       403:
+ *        description: Non admin user trying to fetch other user's connected calendars.
+ */
+async function getHandler(req: NextApiRequest) {
+  const { userId, isAdmin } = req;
+
+  if (!isAdmin && req.query.userId) throw new HttpError({ statusCode: 403, message: "ADMIN required" });
+
+  const userIds = req.query.userId ? extractUserIdsFromQuery(req) : [userId];
+
+  const usersWithCalendars = await prisma.user.findMany({
+    where: { id: { in: userIds } },
+    include: {
+      selectedCalendars: true,
+      destinationCalendar: true,
+    },
+  });
+
+  return await getConnectedCalendars(usersWithCalendars);
+}
+
+async function getConnectedCalendars(users: UserWithCalendars[]) {
+  const connectedDestinationCalendarsPromises = users.map((user) =>
+    getConnectedDestinationCalendars(user, false, prisma).then((connectedCalendarsResult) =>
+      connectedCalendarsResult.connectedCalendars.map((calendar) => ({
+        userId: user.id,
+        ...calendar,
+      }))
+    )
+  );
+  const connectedDestinationCalendars = await Promise.all(connectedDestinationCalendarsPromises);
+
+  return connectedDestinationCalendars.flat();
```

Comment:
We shouldn't return the result as it is, but be specific about what we want to return. I think it's a good idea to place a zod validation (in `api/v1/lib/validation/*`) that we can parse this result in prior to returning it, and only send through data that is necessary. 

I think @joeauyeung would know more about exactly which fields are necessary to be returned, but I doubt it's all of them. Event if it's all of them, thinking about scalability and potential issues in the future (if we add a column that we really should't return), the suggested method of validation would ensure it isn't returned, while the current implementation would return it as it is.

Reply:
Great recommendation! @joeauyeung let me know what properties should be in the response, and I will create zod schema for it ASAP.

Reply:
DM'd @supalarry with the ideal response a customer is looking for.